README.md

```diff
@@ -190,6 +190,38 @@ type Options struct {
 }
 ```
 
+### Custom validator
+
+You can override `options-gen` validator for specific struct by implementing the `Validator()` method:
+```go
+import "github.com/mycoolmodule/internal/validator"
+
+// ...
+
+func (Options) Validator() *validator.Validate {
+  return validator.Validator
+}
+```
+
+Or you can override `options-gen` validator globally:
+```go
+package validator
+
+import (
+  goplvalidator "github.com/go-playground/validator/v10"
+  optsValidator "github.com/kazhuravlev/options-gen/pkg/validator"
+)
+
+var Validator = goplvalidator.New()
+
+func init() {
+  must(Validator.RegisterValidation(/* ... */))
+  must(Validator.RegisterAlias(/* ... */))
+
+  optsValidator.Validator = Validator
```

Comment:
I suggest a method to set validator by public function. This function is accept an interface `Validator`, like `validator.validatorProvider`.

This will allow us to provide an API to change the validator explicitly. And (may be) wrap it with sync.Once.

Reply:
I used this approach first (thoughts of geniuses converge ))), but after met the next concerns:

1. Validator interface is very specific. Can u imagine any other library, that has `Var(field any tag string) error`?
2. There is no reasons to change global validator concurrently or not from `init()` code. And `validator.Validate` object is concurrent safe. Therefore, following the Golang best practices, I decided to simplify the API and just leave the public variable.

What do u think?

P.S. Please pay attention
https://github.com/kazhuravlev/options-gen/pull/35#discussion_r984828083

Reply:
I think my explanation was not good.

All that we need is to set a specific instance of `go-playground/validator`. So, as we use only this validator - we need a function that allows us to set a `*validator.Validate` (concrete instance of validator). We do not need an interface like you mention.

Also, I think that something can be changed in future and function like `SetValidator(v *validator.Validate)` can be transform to (for example) `SetValidator(v ValidatorInterface)`. Also, with the function, we have a super option - print the warning about deprecation.

Reply:
ok, imagine

```go
var validator = goplvalidator.New()

func Set(v *goplvalidator.Validate) {
    validator = v
}
```

but anyway I can do hack like this
```go
v := GetValidatorFor(nil)
v.Register(...) // Working with private global validator
```

okaaaay

```go
var validator IValidator = goplvalidator.New()

type IValidator interface {
    Var(field any, tag string) error
}

func Set(v IValidator) {
    validator = v
}

func GetValidatorFor(opts any) IValidator
```

but anyway my hack is working
```go
v := GetValidatorFor(nil)
v.(*validate.Validate).Register(...) // Working with private global validator
```

---

Maybe we should do some steps forward (or backward) and replace
```go
validator461e464ebed9.GetValidatorFor(o).Var(o.s3Endpoint, "required,url")
```
with
```go
validator461e464ebed9.Validate(o.s3Endpoint, "required,url")
```
but returns to problems about setting and configuring custom validation.

---

It seems to me that there is no way to avoid a strict dependency on this library, and it dictates a global nature and thus puts spokes in our wheels.

Reply:
I suppose that I was not so clear. All that we need: 
- is accept goplvalidator as instance (no interfaces)
- add a function like you wrote `func Set(v *goplvalidator.Validate)`, which will set a private validator variable.

That's all. If someone will do this `GetValidatorFor(nil).Register(...)` - it is ok, because that changes will affect only this project. This is a purpose of setting global custom validator.  It is like http.DefaultClient which you can replace from several packages.

Reply:
@kazhuravlev done